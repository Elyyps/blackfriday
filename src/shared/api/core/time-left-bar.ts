export class TimeLeftBar {
  public value: number;
  public text: number;
}
