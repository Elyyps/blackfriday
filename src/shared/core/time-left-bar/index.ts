export * from "./time-left-bar.component";
