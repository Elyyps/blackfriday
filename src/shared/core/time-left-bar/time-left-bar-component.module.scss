.green {
  color: $brand-secondary;
  .time-left-bar-control-filled {
    background: $brand-secondary;
  }
}
.orange {
  color: #ff9d0a;
  .time-left-bar-control-filled {
    background: #ff9d0a;
  }
}

.time-left-bar {
  color: $gray-lighter;
  margin-bottom: 8px;
  order: 3;
  width: 100%;
  clear: both;
  margin-top: 2px;
  position: relative;

  &-control {
    position: relative;
    height: 3px;
    background: $tint-gray-light;
    border-radius: 3px;
    overflow: hidden;
  }
  &-control-filled {
    position: absolute;
    left: 0;
    top: 0;
    height: 3px;
    background: $tint-gray-light;
  }
}

.responsive {
  order: 3;
  width: 100%;
  clear: both;
  margin-top: 2px;
  position: relative;
  &__status {
    order: 3;
    width: 100%;
    clear: both;
    margin-top: 16px;
    margin-bottom: 16px;
    position: relative;
    &-title {
      position: absolute;
      left: 0;
      top: 28px;
      width: 50%;
    }
  }
}
