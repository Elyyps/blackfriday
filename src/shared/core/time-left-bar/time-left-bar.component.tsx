import * as React from "react";
import styles from "./time-left-bar-component.module.scss";

export interface ITimeLeftBarComponentProps {
  color?: string;
  range: number;
}

const TimeLeftBarComponent = (props: ITimeLeftBarComponentProps) => {
  const { range, color } = props;
  const statusBarFilled = {
    width: `calc( ${range} * 100%)`,
    backgroundColor: color
  };

  return (
    <div className={`${styles["time-left-bar"]}`}>
      <div className={styles["time-left-bar-control"]}>
        <div className={styles["time-left-bar-control-filled"]} style={statusBarFilled} />
      </div>
    </div>
  );
};
TimeLeftBarComponent.defaultProps = {
  color: "none"
};
export { TimeLeftBarComponent };
