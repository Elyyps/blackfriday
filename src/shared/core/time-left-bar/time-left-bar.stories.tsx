import React from "react";
import { shallow } from "enzyme";
import { TimeLeftBarComponent } from "./time-left-bar.component";

describe("[TimeLeftBar]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(
      <TimeLeftBarComponent title="NU GELDIG" range={0.6} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
