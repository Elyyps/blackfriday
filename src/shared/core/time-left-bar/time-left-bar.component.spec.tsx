import React from "react";
import { shallow } from "enzyme";
import { TimeLeftBarComponent } from "./time-left-bar.component";

describe("[TimeLeftBar]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<TimeLeftBarComponent range={0} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
