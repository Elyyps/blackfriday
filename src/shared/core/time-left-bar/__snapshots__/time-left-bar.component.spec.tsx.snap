// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[TimeLeftBar] should render component without crashing 1`] = `
<div
  className="time-left-bar"
>
  <div
    className="time-left-bar-control"
  >
    <div
      className="time-left-bar-control-filled"
      style={
        Object {
          "backgroundColor": "none",
          "width": "calc( 0.5 * 100%)",
        }
      }
    />
  </div>
</div>
`;
